homeassistant/components/epic_games_store/coordinator.py

```diff
@@ -0,0 +1,89 @@
+"""The EGS integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from epicstore_api import EpicGamesStoreAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_LOCALE, DOMAIN
+from .helper import format_game_data, get_country_from_locale
+
+SCAN_INTERVAL = timedelta(days=1)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EGSUpdateCoordinator(DataUpdateCoordinator[dict[str, list[dict[str, Any]]]]):
+    """Class to manage fetching data from the Epic Game Store."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize."""
+        self._api = EpicGamesStoreAPI(
+            entry.data[CONF_LOCALE], get_country_from_locale(entry.data[CONF_LOCALE])
+        )
+        self.locale = entry.data[CONF_LOCALE]
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+
+    async def _async_update_data(self) -> dict[str, list[dict[str, Any]]]:
+        """Update data via library."""
+        try:
+            # pylint: disable-next=protected-access
+            self._api._get_errors = not_handle_service_errors
+            data = await self.hass.async_add_executor_job(self._api.get_free_games)
+        except Exception as error:
```

Comment:
We may only catch broad Exceptions in the config flow. We need to be more specific here

Reply:
Why ?
In what goal ?

We are just getting the data here, and for any reason the data is not fetch we should `raise UpdateFailed(error)` so the coordinator knows the fetch has failed.

Reply:
Let me rephrase it: if we can specify why it failed, we should do that. If we can't, I think the coordinator will catch it itself

Reply:
Fixed: https://github.com/home-assistant/core/commit/2fd933716153266b04592219a387ac5de2e1ab01